### PR TITLE
PS-4775 [MyRocks] Improve documentation about rocksdb_wal_recovery mode

### PR DESCRIPTION
**Pull request by patrickbirch**
Added myrocks/recover_from_crash.rst and updated index.rst

### CODE REVIEW

File: doc/source/myrocks/recover_from_crash.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
+.. _myrocks_recovery:
+
+======================================
+Recovering from a MyRocks Server Crash
+======================================
+
+MyRocks provides multiple crash recovery configurations. Crash recovery is
+accomplished by replaying the transactions from the WAL (Write Ahead Log) files.
+
+You can set the following crash recovery modes in my.cnf to define how WAL files are applied:
```
Comment (george-lorch) [inline doc/source/myrocks/recover_from_crash.rst]: These are for the rocksdb variable wal_recovery_mode.  This variable is set from the MyRocks variable rocksdb_wal_recovery_mode, so, we do not need to discuss the wal_recovery_mode for rocksdb as it is redundant with the description of rocksdb_wal_recovery_mode MyRocks variable.  The original bug report was that our documentation of rocksdb_wal_recovery_mode just states that the value can be from 0 to 3 with no description of what 0, 1, 2, or 3 means.

**Review by george-lorch — CHANGES_REQUESTED** [review]

File: doc/source/myrocks/recover_from_crash.rst (hunk 1 above)
Comment (patrickbirch) [reply doc/source/myrocks/recover_from_crash.rst]: Fixed. I moved some information to the variables page. Removed the Recover page.

**Review by george-lorch — APPROVED** [review]
Some spurious reformatting, but it's OK I guess.

**Comment by patrickbirch** [comment]
Yes. I must investigate why vim is reformatting other areas of the doc.

**Pull request merged**